Chore/refactor-agent-templates

anandxashokan opened the pull request:
1. Cleaned up unnecessary code from custom_react_agent. 
2. Added files to prebuilt_react_agent.
3. Both Agents are functional and can be executed via make command

anandxashokan committed:
update the custom react agent template. Remove the standard react agent which is not used.

anandxashokan committed:
Added a prebuilt react agent template

S1LV3RJ1NX commented on templates/custom-react-agent/src/core/agents/__init__.py:
```diff
@@ -1,4 +1,5 @@
 from src.core.agents.model_provider import MODEL, TOOL_ENABLED_MODEL
 
-__all__ = ["run_agent", "MODEL", "TOOL_ENABLED_MODEL"]
+__all__ = ["run_agent", "run_custom_react_agent", "MODEL", "TOOL_ENABLED_MODEL"]
```
Why can't `run_agent` be used? Instead of `run_custom_react` agent? Since the code is now separated, let's directly use `run_agent` as it calls `run_custom_react` agent

S1LV3RJ1NX commented on templates/custom-react-agent/src/core/agents/agent_factory.py:
use `run_agent` as it calls `run_custom_react` agent

S1LV3RJ1NX replied:
Resolved.

S1LV3RJ1NX commented on templates/custom-react-agent/src/core/agents/agent_factory.py:
```diff
@@ -1,9 +1,4 @@
-# Cache for imported agent functions
-_AGENT_CACHE = {}
-
+from .custom_react_agent import run_custom_react_agent
```
Use absolute imports

S1LV3RJ1NX commented on templates/prebuilt-react-agent/src/core/agents/__init__.py:
```diff
@@ -0,0 +1,5 @@
+from src.core.agents.standard_react_agent import run_react_agent
+from src.core.agents.model_provider import MODEL, TOOL_ENABLED_MODEL
+
+__all__ = ["run_agent", "run_react_agent", "MODEL", "TOOL_ENABLED_MODEL"]
```
Use `run_agent` rather than calling `run_react_agent`, move the implmentation to `run_agent`

S1LV3RJ1NX commented on templates/prebuilt-react-agent/src/core/agents/agent_factory.py:
```diff
@@ -0,0 +1,21 @@
+        dict: A dictionary containing the AI's response.
+    """
+    
+    return await run_react_agent(user_input)
```
move the implementation of `run_react_agent` here

S1LV3RJ1NX commented on templates/prebuilt-react-agent/src/core/agents/model_provider.py:
```diff
@@ -0,0 +1,14 @@
+
+# Initialize base model
+MODEL = ChatOpenAI(
+    model="gpt-4o-mini",
```
later we can add model in the yaml as well, instead of hardcoding it here.

S1LV3RJ1NX commented on templates/prebuilt-react-agent/src/core/agents/model_provider.py:
```diff
@@ -0,0 +1,14 @@
+MODEL = ChatOpenAI(
+    model="gpt-4o-mini",
+    temperature=0,
+    base_url=Settings().LITELLM_GATEWAY_URL,
```
No need to create setting object always. Create the settings object once in the config file and use it everywhere. There is already a `settings` variable present. 
`from src.config import settings`

S1LV3RJ1NX commented on templates/prebuilt-react-agent/src/tools/tool_registry.py:
```diff
@@ -0,0 +1,74 @@
+            - tools_by_name (dict): A dictionary mapping tool names to their instances.
+    """
+    # Get the configuration from settings
+    config = Settings().AGENT_CONFIG
```
Do not create `Settings` object, import `settings` variable.

S1LV3RJ1NX reviewed (changes requested):
Minor issues to fix:
- Avoid calling a function from another function, where first function does nothing but only calls the second function. Rather move the implementation up. Example: move implementations of `run_react_agent` to `run_agent`, etc
- Do not create multiple objects of `Settings` class, the config file already creates the object and assigns it to the `settings` variable, use that instead.

S1LV3RJ1NX committed:
1. Using single instance of Settings across a graph template
2. Changed  tool to  with attributes.
3. Wrote a graph builder class for pre-built-react agent
4. All agent factory now have a single  function

S1LV3RJ1NX reviewed (approved):
Approved.
